tls/s2n_server_new_session_ticket.c

```diff
@@ -44,6 +44,10 @@
 #define S2N_TLS13_MAX_FIXED_NEW_SESSION_TICKET_SIZE 79
 
 int s2n_server_nst_recv(struct s2n_connection *conn) {
+    if (!conn->config->use_tickets) {
+        return S2N_SUCCESS;
+    }
```

Comment:
I don't think you can make this same check in TLS1.2. In TLS1.2, if this message is received, it's received because of the handshake type. It should never be a no-op.